ResearchKit/ActiveTasks/ORKGoNoGoViewController.m

```diff
@@ -110,6 +99,31 @@ - (void)viewWillDisappear:(BOOL)animated {
     _shouldIndicateFailure = NO;
 }
 
+- (NSArray<NSNumber*>*)createRandomArray:(int)size {
```

Comment:
So, previously, you created an array where each item had a 2/3 chance of being YES and 1/3 NO, then checked that it had at least one NO, and if not (and there was more than one test)  it set one of the members of the array other than the 1st to NO.

Now, you choose a random number between 0 and 1/3 the size of the final array,  add 1 (guaranteeing there is now always at least one NO in the array), build an array with that many NOs and filling the rest with YES, then shuffling by iterating once through the array, exchanging each item with one later in the list.

I see two main differences between the two methods. The first is that the NOs in the array no longer appear to be uniformly distributed, and there's a slight bias toward having them in the earlier positions (as far as I can tell.) The other is that in the former you were guaranteed that if there was only one NO in the array, it wouldn't be the first one, and in the latter that's no longer guaranteed.

Is that what you intended?